### ISSUE TITLE
Session timeout not restored on serialized sessions

### ISSUE THREAD

**Issue by Gekkio** [Atmosphere/atmosphere#555]
Atmosphere removes the session timeout when an asynchronous request is started. This timeout is restored if the request stops for some reason (timeout, completed, resumed). This is done by calling `SessionTimeoutSupport.restoreTimeout` in AsynchronousProcessor.

However, if a session is serialized while an asynchronous request is active, the timeout will never be restored after deserialization because the original async request is lost and thus `SessionTimeoutSupport.restoreTimeout` will never be called (unless of course a user returns and starts another comet request with the same session).

This is a pretty bad thing, because by default Tomcat serializes sessions to disk, so in default Tomcat installations Atmosphere apps will always leak sessions that have active asynchronous requests.

One way to fix this is make SessionTimeoutRestorer implement HttpSessionActivationListener, and restore the timeout in sessionDidActivate callback. I think this won't have any negative side-effects because no asynchronous requests are active after deserialization, but I'm not 100% sure about this.

I've tested this with Tomcat 7.0.23 and 7.0.29, and Atmosphere 0.9.7.

**TLDR:**
- Start long comet request in browser
- Stop Tomcat
- Close browser
- Start Tomcat

= Session timeout is not restored and session is leaked permanently unless the session is accessed again or SESSIONS.SER is removed manually.
